src/hooks/mongodb4x.ts

```diff
@@ -32,26 +48,51 @@ const injectMonitoringCommand = (args: any[]) => {
           );
       }
   }
+  return args;
 };
 
-export const beforeConstructorHook = (args: any[], extenderContext: any) => {
-  injectMonitoringCommand(args);
-};
-
-const isMonitoringEnabled = (args: any[]) => {
-  return args.length > 1 && typeof args[1] === 'object' && args[1].monitorCommands === true;
-};
-
-export const afterConstructorHook = (args: any[], clientInstance: any, extenderContext: any) => {
-  if (isMonitoringEnabled(args)) {
-    try {
-      clientInstance.on('commandStarted', safeExecute(onStartedHook));
-      clientInstance.on('commandSucceeded', safeExecute(onSucceededHook));
-      clientInstance.on('commandFailed', safeExecute(onFailedHook));
-    } catch (err) {
-      logger.warn(`MongoDB 4.x 'on' hooks cannot be applied to ${clientInstance}`, err);
-    }
-  } else {
-    logger.debug("MongoDB 4.x 'on' hooks skipped: monitorCommands was not set in the options");
+export const wrapMongoClient4xClass = (mongoClientLibrary: any) => {
+  try {
+    const originalStaticConnect = mongoClientLibrary.MongoClient.connect;
+    const staticProperties = getStaticProperties(mongoClientLibrary.MongoClient);
+    const wrapper = (originalFn: any) => {
+      const wrappedClass = function (...args: any[]) {
```

Comment:
If all the other functions are arrow functions, it's better to keep the inner function as arrow functions as well.

Reply:
I think that it can't be an arrow function. arrow function inherit the scope, and function is not